core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -543,6 +543,11 @@ package object config {
       .version("1.2.0")
       .fallbackConf(DYN_ALLOCATION_SCHEDULER_BACKLOG_TIMEOUT)
 
+  private[spark] val LEGACY_LOCALITY_WAIT_RESET =
+    ConfigBuilder("spark.locality.wait.legacyResetOnTaskLaunch")
```

Comment:
let's make it an internal conf and add the version info.

Reply:
what all does adding internal do? I see internal ones are not exposed when SQLConf.getAllDefinedConfs is called.
Which version should I add?

Reply:
internal means we can remove it later. Ideally all legacy configs should be internal.

we can set version as `3.0.0` and change it if we can't merge it before the release eventually.

Reply:
yes please add the .doc and .version - it would be nice to get into 3.0.0.
I don't know that I agree with the internal tag. Maybe its being used differently then I thought. Internal to me means it's not meant for end user, this one could be if users find issues with the new algorithm. I'm definitely fine with not documenting it and it seems it has that side affect so maybe not a big deal.

Reply:
all the configs are end-user facing (they can set and it will take effect). Internal config is for special use cases only (most of the users should not set it).